src/components/Filters/index.tsx

```diff
@@ -66,482 +29,281 @@ const FilterWrapper = styled(Box)`
-			...view,
-			filters: filters.map(filter => SchemaSieve.flattenSchema(filter)),
-		}));
+const shouldRenderComponent = (
+	mode: FilterRenderMode,
+	renderMode?: FilterRenderMode | FilterRenderMode[],
+): boolean => {
+	// If a render mode is not specified, render all components
+	if (!renderMode) {
+		return true;
 	}
 
-	public emitViewsUpdate() {
-		if (!this.props.onViewsUpdate) {
-			return;
-		}
+	const allowedModes = castArray(renderMode);
 
-		this.props.onViewsUpdate(
-			this.state.views.map(({ filters, ...view }) => ({
-				...view,
-				filters: filters.map(filter => SchemaSieve.unflattenSchema(filter)),
-			})),
-		);
+	if (includes(allowedModes, 'all')) {
+		return true;
 	}
 
-	public getCleanEditModel(field?: string | null) {
-		const schema = this.state.schema;
-		if (!field) {
-			field = Object.keys(schema.properties!).shift()!;
-		}
-
-		const fieldOperators = this.getOperators(field);
-		if (!fieldOperators.length) {
-			return {
-				field,
-				operator: '',
-				value: '',
-			};
-		}
-
-		const operator = fieldOperators.shift()!.slug;
-
-		let value: any = '';
-		const subschema = schema.properties![field];
-		if (typeof subschema !== 'boolean') {
-			if (subschema.enum) {
-				value = subschema.enum[0] || '';
-			}
-
-			if (subschema.oneOf) {
-				value = (subschema.oneOf[0] as JSONSchema6).const || '';
-			}
-
-			if (subschema.type === 'boolean') {
-				value = true;
-			}
-		}
-
-		return {
-			field,
-			operator,
-			value,
-		};
-	}
-
-	public getOperators(field: string) {
-		const schema = this.state.schema;
-		return SchemaSieve.getOperators(schema, field);
-	}
+	return includes(allowedModes, mode);
+};
 
-	public setEditField(field: string, index: number) {
-		const currentEdit = this.state.edit.slice();
-		currentEdit.splice(index, 1, this.getCleanEditModel(field));
-		this.setState({
-			edit: currentEdit,
+const Filters = ({
+	schema,
+	disabled,
+	viewScopes,
+	onFiltersUpdate,
+	onViewsUpdate,
+	addFilterButtonProps,
+	viewsMenuButtonProps,
+	renderMode,
+	dark,
+	compact,
+	...restProps
+}: FiltersProps) => {
+	// Since default values does not work by passing null as a prop value,
+	// I declare them here like so to avoid many checks
+	const filters = restProps.filters ? restProps.filters : [];
+	const views = restProps.views ? restProps.views : [];
+
+	const flatSchema = SchemaSieve.flattenSchema(schema);
+	const flatViews = flattenViews(views);
+	const flatFilters = filters.map(filter => SchemaSieve.flattenSchema(filter));
+	const shouldHideLabel = useBreakpoint(compact || [false]);
+
+	const [filtersState, setFiltersState] = useState<FiltersState>({
+		showModal: false,
+		searchString: '',
+		editingFilter: null,
+		filters: flatFilters,
+		views: flatViews,
+		schema: flatSchema,
+		edit: [],
+	});
+
+	React.useEffect(() => {
+		filtersState.edit.push(SchemaSieve.getCleanEditModel(filtersState.schema));
+	}, []);
+
+	React.useEffect(() => {
+		setFiltersState({
```

Comment:
In order not to add `filtersState` as a dependency and recalculate this any time the state changes, we need to do `setState(filtersState => ...)` instead, otherwise we won't be using the altest filters state.